src/plugins/region_map/public/region_map_type.js

```diff
@@ -36,10 +36,68 @@ import { RegionMapOptions } from './components/region_map_options';
 import { truncatedColorSchemas } from '../../charts/public';
 import { Schemas } from '../../vis_default_editor/public';
 import { ORIGIN } from '../../maps_legacy/public';
+import { getServices } from './services';
+import { DEFAULT_MAP_CHOICE } from '../common';
 
 export function createRegionMapTypeDefinition(dependencies) {
-  const { uiSettings, regionmapsConfig, getServiceSettings, additionalOptions } = dependencies;
+  const {
+    http,
+    notifications,
+    uiSettings,
+    regionmapsConfig,
+    getServiceSettings,
+    additionalOptions,
+  } = dependencies;
   const visualization = createRegionMapVisualization(dependencies);
+  const services = getServices(http);
+
+  const diffArray = (arr1, arr2) => {
+    return arr1.concat(arr2).filter((item) => !arr1.includes(item) || !arr2.includes(item));
+  };
+
+  const getCustomIndices = async () => {
+    try {
+      const result = await services.getCustomIndices();
+      return result.resp;
+    } catch (e) {
+      return false;
+    }
+  };
+
+  const getJoinFields = async (indexName) => {
+    try {
+      const result = await services.getIndexMapping(indexName);
+      const properties = diffArray(Object.keys(result.resp[indexName].mappings.properties), [
+        'location',
+      ]);
+      return properties.map(function (property) {
+        return {
+          type: property,
+          name: property,
+          description: property,
```

Comment:
Curious why all three fields have the same value. Seems redundant

Reply:
Join fields has a structure as follows:

```
description: "ISO 3166-1 alpha-2 Code"
name: "iso_3166_2"
type: "id"
```

Since these are part of custom map, I thought of keeping the name and description the same as property in the document. I'll update type to "id". Let me know if that sounds good.

Reply:
That works. was just curious why 3 fields had the same value.